src/Http/Routing/src/Builder/MinimalActionEndpointRouteBuilderExtensions.cs

```diff
@@ -184,6 +187,21 @@ public static class MinimalActionEndpointRouteBuilderExtensions
             // Add MethodInfo as metadata to assist with OpenAPI generation for the endpoint.
             builder.Metadata.Add(action.Method);
 
+            // Methods defined in a top-level program are generated as statics so the delegate
+            // target will be null. Inline lambdas are compiler generated properties so they can
+            // be filtered that way.
+            if (action.Target == null || !TypeHelper.IsCompilerGenerated(action.Method.Name))
+            {
+                if (!GeneratedNameParser.TryParseLocalFunctionName(action.Method.Name, out var endpointName))
+                {
+                    endpointName = action.Method.Name;
+                }
+
+                builder.Metadata.Add(new EndpointNameMetadata(endpointName));
+                builder.Metadata.Add(new RouteNameMetadata(endpointName));
+                builder.DisplayName = $"{builder.DisplayName} => {endpointName}";
```

Comment:
Missing test(s) for the new display name format?
I don't see any with the form
`HTTP: GET /hello => HelloWorld`

Reply:
Good note! I modified the existing tests but not the new ones. Will add `DisplayName` checks to the new tests.